subsys/mgmt/mcumgr/transport/src/smp_udp.c

```diff
@@ -133,14 +164,78 @@ static int smp_udp_ud_copy(struct net_buf *dst, const struct net_buf *src)
 	return MGMT_ERR_EOK;
 }
 
+static int create_socket(enum proto_type proto)
+{
+	int sock;
+	int err;
+	struct sockaddr *addr;
+
+#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV4
+	struct sockaddr_in addr4;
+#endif
+
+#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV6
+	struct sockaddr_in6 addr6;
+#endif
+
+#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV4
+	if (proto == PROTOCOL_IPV4) {
+		memset(&addr4, 0, sizeof(addr4));
+		addr4.sin_family = AF_INET;
+		addr4.sin_port = htons(CONFIG_MCUMGR_TRANSPORT_UDP_PORT);
+		addr4.sin_addr.s_addr = htonl(INADDR_ANY);
+		addr = (struct sockaddr *)&addr4;
+	}
+#endif
+
+#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV6
+	if (proto == PROTOCOL_IPV6) {
+		memset(&addr6, 0, sizeof(addr6));
+		addr6.sin6_family = AF_INET6;
+		addr6.sin6_port = htons(CONFIG_MCUMGR_TRANSPORT_UDP_PORT);
+		addr6.sin6_addr = in6addr_any;
+		addr = (struct sockaddr *)&addr6;
+	}
+#endif
+
+	sock = socket(addr->sa_family, SOCK_DGRAM, IPPROTO_UDP);
+	err = errno;
+
+	if (sock < 0) {
+		LOG_ERR("Could not open receive socket (%s), err: %i",
+			smp_udp_proto_to_name(proto), err);
+
+		return -err;
+	}
+
+	if (bind(sock, addr, sizeof(*addr)) < 0) {
+		err = errno;
+		LOG_ERR("Could not bind to receive socket (%s), err: %i",
+			smp_udp_proto_to_name(proto), err);
+
+		close(sock);
+
+		return -err;
+	}
+
+	return sock;
+}
+
 static void smp_udp_receive_thread(void *p1, void *p2, void *p3)
 {
 	struct config *conf = (struct config *)p1;
 
 	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
-	LOG_INF("Started (%s)", conf->proto);
+	(void)k_sem_take(&conf->network_ready_sem, K_FOREVER);
```

Comment:
This could be potentially removed, since in any case when the network disconnects, this thread will be waiting on `recvfrom()` until a new connection occurs.
```suggestion
```

Reply:
If this suggestion is accepted, there is no need for the semaphore itself, ie. `network_ready_sem`

Reply:
This waits for the network interface to come up, the thread being started does not indicate that the network is up

Reply:
So, what happens when the network disconnects?

Reply:
It works. The point in waiting here is to support network interfaces that are not registered at startup and get registered whilst an application is running (functionality not tested as I don't have any boards that do this)

Reply:
Good point.
This is also a good synchronization point in `case smp_udp_open()` is called by application/initialisation code before `smp_udp_start()`